source/bao_hyp/user_manual/porting.rst

```diff
@@ -0,0 +1,270 @@
+and **regions** fields. The former is the number of available regions while the latter defines an array of `struct mem_region` with the following fields:
+
+.. code-block:: c
+
+    struct mem_region {
+        paddr_t base;
+        size_t size;
+        ...
+    };
+
+Where:
+
+- **base** [mandatory] - Is the base physical address of that memory region. Must be aligned to the minimum architecture's page size;
+- **size** [mandatory] - The size of the memory region. Must be aligned to the minimum architecture's page size;
+
+Note that, by convention, we do not add on-chip/scratcpad SRAMs as a platform memory region as the hypervisor as an uniform view of memory.
+This regions should be assigned to guests in the form of devices.
+
+
+Cache Description
+#################
+
+TODO
+
+
+Architcture-Specific Platform Description
+#########################################
+
+The architecture-specific field of the platform description includes fields that describe interrupt controllers or others. 
+
+.. tabs::
+    .. tab:: Arm
+
+    For the Arm architecure, it includes the following fields:
+
+    .. code-block:: c
+
+        struct arch_platform {
+            struct gic_dscrp {
+                paddr_t gicc_addr;
+                paddr_t gich_addr;
+                paddr_t gicv_addr;
+                paddr_t gicd_addr;
+                paddr_t gicr_addr;
+
+                irqid_t maintenance_id;
+            } gic;
+
+            struct smmu_dscrp {
+                paddr_t base;
+                streamid_t global_mask;
+            } smmu;
+
+            struct clusters {
+                size_t num;
+                size_t* core_num;
+            } clusters;
+        };
+
+    Where, for the GIC interrupt controller `struct gic_dscrp` description:
+
+    - **gic.gicc_addr** [mandatory for GICv2 platforms] - base address for the GIC's CPU Interface;
+    - **gic.gich_addr** [mandatory for GICv2 platforms] - base address for the GIC's Virtual Interface Control Registers;
+    - **gic.gicv_addr** [mandatory for GICv2 platforms] - base address for the GIC's Virtual CPU Interface;
+    - **gic.gicd_addr** [mandatory] - base address for the GIC's Distributor;
+    - **gic.gicr_addr** [mandatory for GICv3/4 platforms] - base address for the GIC's Redistributor;
+    - **gic.maintenance_id** [mandatory] - The interrupt ID for the GIC's maintenance interrupt;
+
+    For the SMMU `struct smmu_dscrp`:
+
+    - **smmu.base** [mandatory] - is the base address for the SMMU;
+    - **smmu.global_mask** [optional; only valid for SMMUv2] - a mask to be applied to all SMMUv2's Stream Match Registers;
+
+
+    Finally, when CPUs are organized in clusters, in the Arm architecture their IDs are assigned using an hierarchical schema.
+    To be able to calculate the linearized ID for each core, we require the port to provide the number of CPUs of cluster in
+    ascending order of AFF1.
+
```

Comment:
This needs an extra indentation to be grouped on each tab. Same for the RISC-V information.